gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/predicates/DatasetHiveSchemaContainsNonOptionalUnion.java

```diff
@@ -77,7 +89,13 @@ private DbAndTable getDbAndTable(T dataset) {
       "Expected pattern = %s", dataset.getUrn(), pattern.pattern()));
     }
 
-    return new DbAndTable(m.group(1), HiveMetaStoreUtils.getHiveTableName(m.group(2)));
```

Comment:
We need to be cautious here with the line above, if the pattern match doesn't work it will throw before it runs the lines below. You may want to avoid throwing until after the secondary check fails

Reply:
If the pattern match doesn't work we won't anyway have the tableName so it is fine for the 2nd check not to run and exit out saying pattern is not matching. 

As the 2nd check is only for the db, whereas the pattern check is for both the db and table name and if no table name, no point having just the optionalDbName present.